metrics/internal/cache/pv.go

```diff
@@ -171,6 +227,12 @@ func (p *PersistentVolumeStore) add(pv *corev1.PersistentVolume) error {
 		return fmt.Errorf("failed to get node name for pod: %v", err)
 	}
 
+	childrenCount, err := p.runCephRBDChildrenCountFn(&p.monitorConfig, pv.Spec.CSI.VolumeAttributes["pool"], pv.Spec.CSI.VolumeAttributes["radosNamespace"], pv.Spec.CSI.VolumeAttributes["imageName"])
+	if err != nil {
+		return fmt.Errorf("failed to get children count for image %s/%s: %v", pv.Spec.CSI.VolumeAttributes["imageName"], pv.Spec.CSI.VolumeAttributes["radosNamespace"], err)
+	}
+	p.RBDChildrenMap[pv.Spec.CSI.VolumeAttributes["imageName"]] = childrenCount
```

Comment:
Don't you need some sort of mutex to protect against parallel access to the map?

Reply:
while calling this add function to add the pv details, we already have a mutex for hold parallel access to the store itself, I don't think we need separate mutex for individual operations? please do correct me if I am wrong? cc @Madhu-1 @weirdwiz

Reply:
@yati1998 locks need to be as short as possible. This means that it would be preferable to only lock the access to the map itself. Also as @Madhu-1 stated above, it will be preferable if we don't give direct access to the map and only allow access using an accessor that does both a `lock` and a `defer unlock`. 

We also need to separate between reads and writes, as reads can happen in parallel, this is why there is a RLock (read lock) operation in addition to Lock (which is a full lock operation)